napari/_vispy/vispy_shapes_layer.py

```diff
@@ -29,6 +29,8 @@ def _on_data_change(self):
         faces = self.layer._data_view._mesh.displayed_triangles
         colors = self.layer._data_view._mesh.displayed_triangles_colors
         vertices = self.layer._data_view._mesh.vertices
+        if vertices is not None:
+            vertices = vertices + 0.5
```

Comment:
I'm concerned that this is repeated in many places... Should this not be fixed higher upstream?

Reply:
it would be a minor code change to vispy, but could cause some backwards compatibility issues for them. There might be a better way to do it that involves offsetting the image visual instead of all the mesh visuals, but this works for now

Reply:
This is such an ingrained way of doing things in OpenGl, I doubt that this is something that would be called a 'fix'. As we discussed, it is a convention. For our purpose it makes sense to have it 'fixed'. Please keep in mind that the drawback of this convention is that some parts of the image have negative coordinates... (for example, 75% of pixel at (0,0) has negative coordinates)